### PR TITLE
[ADD] estate: Add new features to offer, area and best price fields T#70778

### PR DESCRIPTION
**Pull request by Gilmaciasr**
This PR contains changes made according to chapter 9 from odoo getting started.
Link to the lesson:
https://www.odoo.com/documentation/15.0/developer/tutorials/getting_started/09_compute_onchange.html
Video of the changes:
https://youtu.be/iP9t6-fuJ2M
https://youtu.be/LUs6vxDewbs

### CODE REVIEW

File: estate/views/estate_property_views.xml [hunk 1]
```diff
@@ -54,18 +55,21 @@
                   <field name="garden"/>
                   <field name="garden_area" string="Garden Area (sqm)"/>
                   <field name="garden_orientation"/>
+                  <field name="total_area" string="Total Area (sqm)"/>
                 </group>
               </page>
               <page string="Offers">
-                <field name="offer_ids">
+                <field name="offer_ids" string="Offer">
```
Comment (JesusValdez96) [inline estate/views/estate_property_views.xml:62]: I think that odoo converts that automatically

File: estate/views/estate_property_views.xml [hunk 2]
```diff
@@ -54,18 +55,21 @@
...
                     <field name="price"/>
                     <field name="partner_id"/>
+                    <field name="validity" string="Validity (days)"/>
+                    <field name="date_deadline" string="Deadline"/>
                     <field name="status"/>
                   </tree>
                 </field>
               </page>
-              <page string="Other info">
+              <page string="Other_info">
```
Comment (JesusValdez96) [inline estate/views/estate_property_views.xml:72]: Why the "_"???

```suggestion
              <page string="Other info">
```

**Review by JesusValdez96 — CHANGES_REQUESTED** [review]

File: estate/models/estate_property_offer.py [hunk 3]
```diff
@@ -23,3 +23,19 @@ class EstatePropertyOffer(models.Model):
         comodel_name='estate.property',
```
[the remark this answers is not in the document]
Comment (Gilmaciasr) [reply estate/models/estate_property_offer.py]: Done.
I also changed some string above that has the same issue.

File: estate/models/estate_property_offer.py [hunk 4]
```diff
@@ -23,3 +23,19 @@ class EstatePropertyOffer(models.Model):
...
+    create_date = fields.Date(default=fields.Date.today())
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+
+    @api.depends("validity", "create_date")
+    def _compute_date_deadline(self):
+        for record in self:
+            record.date_deadline = fields.Date.add(
+                record.create_date, days=record.validity
+            )
```
[the remark this answers is not in the document]
Comment (Gilmaciasr) [reply estate/models/estate_property_offer.py]: Okay!

File: estate/models/estate_property_offer.py [hunk 5]
```diff
@@ -23,3 +23,19 @@ class EstatePropertyOffer(models.Model):
...
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            record.create_date = fields.Date.subtract(
+                record.date_deadline, days=record.validity
+            )
```
[the remark this answers is not in the document]
Comment (Gilmaciasr) [reply estate/models/estate_property_offer.py]: Roger that!

File: estate/views/estate_property_views.xml [hunk 6]
```diff
@@ -5,13 +5,13 @@
...
         <field name="arch" type="xml">
-          <search string="search property">
+          <search string="search_property">
             <field name="name" string="Title"/>
             <field name="property_type_id"/>
             <field name="postcode"/>
             <field name="expected_price"/>
             <field name="bedrooms"/>
-            <field name="living_area" string="Living Area (sqm)"/>
+            <field name="living_area" string="Living_Area_(sqm)"/>
```
[the remark this answers is not in the document]
Comment (Gilmaciasr) [reply estate/views/estate_property_views.xml]: in previous PR you suggest something like that, maybe i got confused.

File: estate/models/estate_property_offer.py [hunk 7]
```diff
@@ -10,16 +10,28 @@ class EstatePropertyOffer(models.Model):
...
+        comodel_name="res.partner",
+        string="Partner",
         required=True,
     )
     property_id = fields.Many2one(
-        comodel_name='estate.property',
+        comodel_name="estate.property",
         required=True,
     )
+    create_date = fields.Date(default=fields.Date.today())
```
Comment (JesusValdez96) [inline estate/models/estate_property_offer.py:26]: The date of fields.Date.today() will always be equal to the date the server was started if you don't use a lambda function.

```suggestion
    create_date = fields.Date(default=lambda self: fields.Date.today())
```

**Review by JesusValdez96 — CHANGES_REQUESTED** [review]
Hi @Gilmaciasr Good job! only two comments.

**Pull request merged**